server/src/main/java/org/elasticsearch/index/cache/query/QueryCacheStats.java

```diff
@@ -23,24 +23,33 @@
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.common.io.stream.Streamable;
+import org.elasticsearch.common.io.stream.Writeable;
 import org.elasticsearch.common.unit.ByteSizeValue;
 import org.elasticsearch.common.xcontent.ToXContent;
 import org.elasticsearch.common.xcontent.ToXContentFragment;
 import org.elasticsearch.common.xcontent.XContentBuilder;
 
 import java.io.IOException;
 
-public class QueryCacheStats implements Streamable, ToXContentFragment {
+public class QueryCacheStats implements Streamable, Writeable, ToXContentFragment {
 
-    long ramBytesUsed;
-    long hitCount;
-    long missCount;
-    long cacheCount;
-    long cacheSize;
+    private long ramBytesUsed;
```

Comment:
I noticed that some classes are using Atomic equivalents of these primitives. Does this mean that these objects could be published to another thread?
@talevy do you have an opinion on this? have a look at `RecoveryStats` where we have fields final atomics

Reply:
I spoke with @talevy and we decided that this is safe to keep them primitives here. The Atomics used in other classes are there because of multithreaded context these classes are used in.